src/util/checkIfPathShareSubcontext.js

```diff
@@ -0,0 +1,24 @@
+export const checkIfPathShareSubcontext = (rankedThoughtsA, rankedThoughtsB) => {
+  const longPath = rankedThoughtsA.length > rankedThoughtsB.length ? rankedThoughtsA : rankedThoughtsB
+  const shortPath = rankedThoughtsB.length < rankedThoughtsA.length ? rankedThoughtsB : rankedThoughtsA
+
+  /* checking if common majoity subcontext is possible */
+  if (shortPath.length < Math.floor(longPath.length / 2)) return { isSubcontext: false, index: -1 }
+
+  const subcontextData = { isSubcontext: false, index: -1 }
```

Comment:
The tuple is unnecessary here since `isSubcontext` is derivable; it is only true when `index` >= 0. Therefore you can simply use `index`.